pkg/instance/service.go

```diff
@@ -313,6 +316,71 @@ func (s service) deployDeploymentInstance(token string, instance *model.Deployme
 	return nil
 }
 
+func (s service) DeleteDeployment(id uint) error {
+	deployment, err := s.FindDeploymentById(id)
```

Comment:
We find the deployment 3 times if I am counting correctly 😅 Since we call `s.instanceRepository.DeleteDeployment(id)` only from here, can we pass in the `Deployment` to not fetch it again?